presto-product-tests/src/main/java/io/prestosql/tests/hive/TestAvroSymlinkInputFormat.java

```diff
@@ -87,12 +72,44 @@ public void testSymlinkTable()
                 "STORED AS " +
                 "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat' " +
                 "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'");
-
-        hdfsClient.delete(warehouseDirectory + "/test_avro_schema_symlink/symlink.txt");
+        hdfsClient.createDirectory(warehouseDirectory + "/TestAvroSymlinkInputFormat/data");
+        saveResourceOnHdfs("avro/original_data.avro", warehouseDirectory + "/TestAvroSymlinkInputFormat/data/original_data.avro");
+        hdfsClient.saveFile(warehouseDirectory + "/TestAvroSymlinkInputFormat/data/dontread.txt", format("This file will cause an error if read as avro."));
         hdfsClient.saveFile(warehouseDirectory + "/test_avro_symlink/symlink.txt", format("hdfs://%s/TestAvroSymlinkInputFormat/data/original_data.avro", warehouseDirectory));
+        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_symlink")).containsExactly(row("someValue", 1));
+        onHive().executeQuery("DROP TABLE test_avro_symlink");
+        hdfsClient.delete(warehouseDirectory + "/TestAvroSymlinkInputFormat");
+    }
 
-        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_symlink"))
-                .containsExactly(row("someValue", 1));
-        onHive().executeQuery("DROP TABLE IF EXISTS test_avro_symlink");
+    @Test(groups = {AVRO, STORAGE_FORMATS})
+    public void testSymlinkTableWithDirectory()
+            throws IOException
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_avro_symlink_with_directory");
+
+        onHive().executeQuery("" +
+                "CREATE TABLE test_avro_symlink_with_directory " +
+                "ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe' " +
+                "WITH SERDEPROPERTIES ('avro.schema.literal'='{" +
+                "\"namespace\": \"io.prestosql.tests.hive\"," +
+                "\"name\": \"test_avro_symlink\"," +
+                "\"type\": \"record\"," +
+                "\"fields\": [" +
+                "{ \"name\":\"string_col\", \"type\":\"string\"}," +
+                "{ \"name\":\"int_col\", \"type\":\"int\" }" +
+                "]}') " +
+                "STORED AS " +
+                "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat' " +
+                "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'");
+
+        String tableRoot = "/TestAvroSymlinkInputFormat";
```

Comment:
make it `warehouseDirectory + "/TestAvroSymlinkInputFormat"`. This way you don't have to add `warehouseDirectory` later.